modules/pagination/client-react/containers/DataProvider.jsx

```diff
@@ -35,8 +35,8 @@ export const useDataProvider = () => {
     },
     [items]
   );
-
-  useEffect(() => {
+  const useMountEffect = fn => useEffect(fn, []);
```

Comment:
Thanks, but why do you create `useMountEffect` here, why just not pass empty array to `useEffect`?

Reply:
ESLint eslint-plugin-react-hooks doesn't allow passing an empty dependency array. It autocorrects it. See https://github.com/facebook/react/issues/15063

Reply:
In this case lets maybe use `// eslint-disable-next-line ...` to disable the eslint rule for this line, since we know this is a valid use case and not the case where we just forgot deps. There is no deps, it should be fired one time.

Reply:
IMHO disabling ESLint rule doesn't clearly demonstrate the developer's intent. It doesn't have a clear explanation or a hint "why". By declaring useMountEffect we are adding a semantic to it (clearly stating that we would like to use it only on mount), and avoiding ESLint rule disabling which is not desirable.

Reply:
Well, declaring useMountEffect is not much better solution as well, since we don't solve the underlying problem, but rather mask it, by introducing additional function wrapper. The underlying problem is that the render code itself is not written cleanly and it should be rewritten instead, IMHO.